### ISSUE TITLE
Kernel 6.6 Support

### ISSUE THREAD

**Issue by AhmadMasry** [bottlerocket-os/bottlerocket#3604]
**What I'd like:**
New Long term Linux kernel 6.6 has been released, I would like to see newer Bottlerocket version to support this kernel version.

**Any alternatives you've considered:**

**Comment by cbgbt** [comment]
Thanks for filing this issue! I'm not sure what the plan is to release a kernel 6.6 variant just yet, but we'll keep track of that here.

**Comment by sumeet-zuora** [comment]
Did realize the kernels on ami k8s 1.27 is still at 5.15 any plans of upgrading to newer? and where can we see the roadmap

**Comment by vyaghras** [comment]
Bottlerocket avoids upgrading the Kernel version for existing variants to avoid making backwards-incompatible changes. If you are looking to use a newer kernel, you can use k8s-1.28 ami that uses "kernel-6.1".

**Comment by sumeet-zuora** [comment]
But current cilum 1.14 does not support 1.28, hence till that time we are on 1.27

**Comment by vyaghras** [comment]
@sumeet-zuora Is there any specific feature you are looking for from Kernel newer than 5.15?

**Comment by sumeet-zuora** [comment]
Yes, one is BBR https://isovalent.com/blog/post/accelerate-network-performance-with-cilium-bbr/ and another good to have is Big TCP https://isovalent.com/blog/post/big-tcp-on-cilium/

**Comment by stockholmux** [comment]
@sumeet-zuora So @vyaghras did a good job going over this, but I did want to add a concrete reference here: our kernel update policy is on the [`SECURITY_FEATURES` document](https://github.com/bottlerocket-os/bottlerocket/blob/develop/SECURITY_FEATURES.md#update-policy).

Looks like cilium will be compatiable with k8s 1.28 in their next release, so after that you can migrate your cluster to 1.28 and bottlerocket to the newer `aws-k8s-1.28` variant which has Linux 6.1 kernel (also also quite few other nice new things).

**Comment by AhmadMS1988** [comment]
Any chance this can be included in with aws-k8s-1.30?

**Comment by yeazelm** [comment]
Hello @AhmadMS1988, we do not have plans to move to 6.6 for `aws-k8s-1.30`. We track the same long term releases as Amazon Linux and the latest they are tracking is 6.1.

**Comment by alex-berger** [comment]
@yeazelm Does that mean, that we will have to wait for [Amazon Linux 2025](https://docs.aws.amazon.com/linux/al2023/ug/release-cadence.html), or even longer, before we can move beyond 6.1?

Is there a realistic chance, Bottlerocket could adopt a more aggressive upgrade policy than Amazon Linux, e.g. one that at least considers the most recent Linux Kernel LTS version?

**Comment by alex-berger** [comment]
How, can we change this, what is the right place to make lobby for decoupling Bottlerocket Linux kernel versioning policy from *Amazon Linux 2023* one? Is it here with the open source Bottlerocket community or does it make more sense to talk to AWS?

**Comment by bcressey** [comment]
The Bottlerocket AMIs published in EC2 - the ones with "owner: amazon" - will likely always be running a kernel that's supported by the Amazon Linux team. There was a fairly long period where that team would release new LTS kernels each year: 4.14, 4.19, 5.4, 5.10, 5.15, 6.1.

6.6 was the first LTS kernel where the upstream kernel.org support went from six years to two years. It was also the first time Amazon Linux skipped an LTS kernel release. That's not surprising, everyone who ships a kernel has had to work through the implications of upstream changes. The firehose of CVEs coming from the new kernel.org CNA is another factor: it's no longer tenable to stay on an LTS kernel after upstream support ends, or at least not for very long.

In the same period EKS announced extended support, meaning that Bottlerocket's Kubernetes variants now need to live for 26 months, instead of 14.

My heartfelt desire is to never change the LTS kernel for a variant once it's shipped, but since 26 months is greater than 24 months, that breaks down even if we assume Amazon Linux ships the LTS kernel instantly and that Bottlerocket instantly makes it available in the next Kubernetes variant, which always comes out the week after the kernel.

Failing that, what I would _like_ to see is a cadence where there's a new LTS kernel every year, a FIPS certification for that kernel a year later, and variants stick to that kernel for the first half of their life, and move to a subsequent kernel for the second, extended support half.

For example:
* Dec 2024: 6.12 LTS kernel released
* Mar 2025: Amazon Linux supports 6.12
* Apr 2025: `aws-k8s-1.33` ships with 6.12, `aws-k8s-1.33-fips` ships with 6.1
*  ... other k8s variants shipped ...
* Dec 2025: 6.18 LTS kernel released
* Mar 2026: Amazon Linux supports 6.18, FIPS certificate released for 6.12 kernel
* Apr 2026: `aws-k8s-1.33` moves to 6.18, `aws-k8s-1.33-fips` moves to 6.12
* ... other k8s variants refresh on their anniversary ...
* ... cycle continues ...

However, I doubt this can be sustained indefinitely since LTS kernel releases, Kubernetes releases, Amazon Linux releases, and FIPS certifications are not actually coordinated. There will eventually be undesirable outcomes like a variant that gets an LTS kernel upgrade the week after it's released, or the month before the variant goes EOL.

What I'd anticipate instead is a policy to refresh kernels across the board annually: every July, all normal variants move to the newer LTS kernel (six months after upstream release), and all FIPS variants move to the newly certified FIPS kernel (18 months after upstream release). That has the advantage of being predictable, but it depends on two factors outside of my control - the Amazon Linux release and the FIPS certification timeline.

**Comment by bcressey** [comment]
The above applies to the "official" AMIs and VMware images.

For custom builds, I'm looking into whether we can add other distro kernels - CentOS Stream, Debian - to the kernel kit. My idea is that these would be updated whenever a new upstream release came out, and would go away when upstream support ended. This would be most useful for non-EC2 bare metal variants but might be interesting for EC2 if caveats like "not covered by paid AWS support plans" did not scare people off.

Someone could also provide a kit with packages for stable kernels, or maybe just the most recent stable kernel release (whatever that happened to be). That would have to be a community effort though, as the risk of regressions would be very quite high and there'd be a lot of work downstream to keep up to date and thoroughly test each new version.

**Comment by alex-berger** [comment]
Looks like starting with [v1.39.0](https://github.com/bottlerocket-os/bottlerocket/releases/tag/v1.39.0) Bottlerocket Linux variant for Kubernetes 1.33 is now using Linux Kernel 6.12 (which is AFAIK the most recent LTS kernel version).
